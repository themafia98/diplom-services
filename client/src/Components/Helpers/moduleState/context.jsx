import { createContext } from 'react';

const ModuleContextState = createContext();
const ModuleContextActions = createContext();

export { ModuleContextState, ModuleContextActions };
