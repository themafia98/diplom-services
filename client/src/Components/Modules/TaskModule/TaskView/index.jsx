import React from "react";
import PropTypes from "prop-types";
import _ from "lodash";
import moment from "moment";
import { Descriptions, Empty, Input, Select, DatePicker, message } from "antd";
import { connect } from "react-redux";
import Scrollbars from "react-custom-scrollbars";

import { getSchema } from "../../../../Utils/index";
import { TASK_SCHEMA } from "../../../../Utils/schema/const";

import { middlewareCaching, middlewareUpdate } from "../../../../Redux/actions/publicActions/middleware";

import uuid from "uuid/v4";

import ModalWindow from "../../../ModalWindow";
import Output from "../../../Output";
import TitleModule from "../../../TitleModule";
import Comments from "../../../Comments";
import File from "../../../File";

const { Option } = Select;

class TaskView extends React.PureComponent {
    state = {
        key: this.props.uuid ? this.props.uuid : null,
        mode: "jur",
        modeControll: "default",
        modeControllEdit: {
            key: null,
            status: null,
            name: null,
            priority: null,
            author: null,
            editor: null,
            description: null,
            date: null
        },
        primaryKey: "___taskViewSetJurnal",
        showModalJur: false,
        modeEditContent: false
    };

    static propTypes = {
        onCaching: PropTypes.func.isRequired,
        onUpdate: PropTypes.func.isRequired,
        height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        onLoadCurrentData: PropTypes.func,
        data: PropTypes.oneOfType([PropTypes.object, () => null])
    };

    static getDerivedStateFromProps = (props, state) => {
        if (props.key !== state.key) return { ...state, key: props.key };
        else return state;
    };

    componentDidMount = () => {
        const {
            publicReducer: { caches = {} } = {},
            router: { routeDataActive: { key = "" } = {}, routeDataActive = {} },
            onCaching,
            data: { key: keyProps = "" } = {}
        } = this.props;
        const { primaryKey: primaryKeyState } = this.state;
        const primaryKey = !_.isEmpty(routeDataActive) && key ? key : keyProps ? keyProps : "";
        if (_.isEmpty(caches) || (key && !caches[key]) || !key) {
            onCaching(null, primaryKey, "GET", primaryKeyState, "jurnalWork");
        }
    };

    onEdit = event => {
        const {
            router: { routeDataActive = {} }
        } = this.props;
        this.setState({
            ...this.state,
            modeEditContent: false,
            modeControll: "edit",
            modeControllEdit: { ...routeDataActive }
        });
    };

    onRejectEdit = event => {
        this.setState({
            ...this.state,
            modeControll: "default",
            modeEditContent: false,
            modeControllEdit: {
                key: null,
                status: null,
                name: null,
                priority: null,
                author: null,
                editor: null,
                description: null,
                date: null
            }
        });
    };

    onChangeEditableStart = event => {
        const dateString = event && event._d ? moment(event._d, "DD.MM.YYYY").format("DD.MM.YYYY") : null;
        const { modeControllEdit: { date = [] } = {} } = this.state;
        const newDate = [...date];
        newDate[0] = dateString;

        return this.setState({
            ...this.state,
            modeEditContent: false,
            modeControllEdit: {
                ...this.state.modeControllEdit,
                date: newDate
            }
        });
    };

    onChangeEditableEnd = event => {
        const dateString = event && event._d ? moment(event._d, "DD.MM.YYYY").format("DD.MM.YYYY") : null;
        const { modeControllEdit: { date = [] } = {} } = this.state;
        const newDate = [...date];
        newDate[1] = dateString;
        return this.setState({
            ...this.state,
            modeEditContent: false,
            modeControllEdit: {
                ...this.state.modeControllEdit,
                date: newDate
            }
        });
    };

    onChangeEditable = event => {
        const { currentTarget = {}, currentTarget: { value = "" } = {} } = event;
        if (_.isObject(event) && currentTarget && !_.isEmpty(currentTarget)) {
            return this.setState({
                ...this.state,
                modeEditContent: false,
                modeControllEdit: {
                    ...this.state.modeControllEdit,
                    name: value
                }
            });
        } else if (_.isObject(event)) {
            if (_.isArray(event)) {
                return this.setState({
                    ...this.state,
                    modeEditContent: false,
                    modeControllEdit: {
                        ...this.state.modeControllEdit,
                        editor: [...event]
                    }
                });
            }
        } else if (typeof event === "string") {
            const arrayStatus = ["Открыт", "Выполнен", "Закрыт", "В работе"];
            if (arrayStatus.some(it => it === event)) {
                return this.setState({
                    ...this.state,
                    modeEditContent: false,
                    modeControllEdit: {
                        ...this.state.modeControllEdit,
                        status: event
                    }
                });
            } else {
                return this.setState({
                    ...this.state,
                    modeEditContent: false,
                    modeControllEdit: {
                        ...this.state.modeControllEdit,
                        priority: event
                    }
                });
            }
        }
    };

    onUpdateEditable = event => {
        const { onUpdate, router: { routeDataActive = {} } = {} } = this.props;
        const { modeControllEdit = {} } = this.state;
        const validHashCopy = [{ ...modeControllEdit }];
        const validHash = validHashCopy.map(it => getSchema(TASK_SCHEMA, it, "no-strict")).filter(Boolean)[0];

        if (validHash)
            onUpdate(
                modeControllEdit.key,
                "UPDATE",
                { ...validHash },
                null,
                { ...routeDataActive },
                "tasks",
                "tasks",
                true
            )
                .then(() => {
                    this.onRejectEdit(event);
                    message.success("Задача обновлена.");
                })
                .catch(error => {
                    message.error("Ошибка обновления задачи.");
                });
    };

    onCancelEditModeContent = event => {
        this.setState({
            ...this.state,
            modeEditContent: false
        });
    };

    onEditContentMode = event => {
        this.setState({
            ...this.state,
            modeEditContent: true
        });
    };

    showModalWorkJur = event => {
        this.setState({
            ...this.state,
            mode: "jur",
            showModalJur: true,
            modeEditContent: false
        });
    };

    renderWorkJurnal = (jurnalDataKeys = []) => {
        const { publicReducer: { caches = null } = {} } = this.props;

        return _.uniq(jurnalDataKeys)
            .map(key => {
                const item = caches[key];
                return (
                    <div key={key} className="jurnalItem">
                        <p>
                            <span className="title">Затрачено времени:</span>{" "}
                            {item && item.timeLost ? item.timeLost : item[0] ? item[0].timeLost : "не установлено"}
                        </p>
                        <p>
                            <span className="title">Дата:</span>{" "}
                            {item && item.date
                                ? moment(item.date).format("DD.MM.YYYY HH:mm:ss")
                                : item[0]
                                ? item[0].date
                                : "не установлено"}
                        </p>
                        <p>
                            <span className="title">Коментарии:</span>
                        </p>
                        <p>
                            {item && item.description
                                ? item.description
                                : Array.isArray(item)
                                ? item[0] && item[0].description
                                    ? item[0].description
                                    : "не установлено"
                                : "не установлено"}
                        </p>
                    </div>
                );
            })
            .filter(Boolean)
            .sort((a, b) => (a.date && b.date ? moment(a.date).valueOf() - moment(b.date).valueOf() : a - b));
    };

    render() {
        const {
            router: { routeDataActive = {} },
            onCaching,
            onUpdate,
            publicReducer: { caches = null } = {},
            path
        } = this.props;
        const { mode, primaryKey, modeControll, modeEditContent, modeControllEdit } = this.state;
        const {
            key = "",
            status = "",
            priority = "",
            name = "",
            author = "",
            editor = [],
            date = [],
            description = ""
        } = routeDataActive || {};
        let jurnalDataKeys = null;
        if (caches && primaryKey && routeDataActive && key) {
            const keys = Object.keys(caches);

            jurnalDataKeys = keys.filter(keyData => keyData.includes(primaryKey) && keyData.includes(key));
        }

        const accessStatus = _.uniq([
            modeControllEdit.status ? status : status ? status : null,
            "Открыт",
            "Выполнен",
            "Закрыт",
            "В работе"
        ]).filter(Boolean);

        const accessPriority = _.uniq([
            modeControllEdit.priority ? modeControllEdit.priority : priority ? priority : null,
            "Высокий",
            "Средний",
            "Низкий"
        ]).filter(Boolean);
        const rulesEdit = true;

        const statusClassName = routeDataActive
            ? status === "Выполнен"
                ? "done"
                : status === "Закрыт"
                ? "close"
                : status === "В работе"
                ? "active"
                : null
            : null;

        if (key) {
            return (
                <React.Fragment>
                    <TitleModule classNameTitle="taskModuleTittle" title="Карточка задачи" />
                    <ModalWindow
                        onCaching={onCaching}
                        primaryKey={primaryKey}
                        routeDataActive={routeDataActive}
                        mode={mode}
                        path={path}
                        typeRequst={"POST"}
                        key={key ? key : uuid()}
                        keyTask={key ? key : null}
                        accessStatus={accessStatus}
                        onUpdate={onUpdate}
                        onEdit={this.onEdit}
                        onRejectEdit={this.onRejectEdit}
                        modeControll={modeControll}
                        editableContent={description}
                        modeEditContent={modeEditContent}
                        onCancelEditModeContent={this.onCancelEditModeContent}
                        onUpdateEditable={this.onUpdateEditable}
                        statusTaskValue={status ? status : null}
                    />
                    <div className="taskView">
                        <div className="col-6 col-taskDescription">
                            <Scrollbars>
                                <Descriptions bordered column={{ xxl: 1, xl: 1, lg: 1, d: 1, sm: 1, xs: 1 }}>
                                    <Descriptions.Item label="Артикул">
                                        <Output className="key">{key}</Output>
                                    </Descriptions.Item>
                                    <Descriptions.Item label="Название">
                                        {modeControll === "default" ? (
                                            <Output className="name">{name}</Output>
                                        ) : modeControll === "edit" && modeControllEdit ? (
                                            <Input
                                                className="nameEdit"
                                                onChange={this.onChangeEditable}
                                                value={modeControllEdit.name}
                                            />
                                        ) : null}
                                    </Descriptions.Item>
                                    <Descriptions.Item label="Статус">
                                        {modeControll === "default" ? (
                                            <Output className={["status", statusClassName].join(" ")}>{status}</Output>
                                        ) : modeControll === "edit" && modeControllEdit ? (
                                            <Select
                                                className="statusEdit"
                                                value={modeControllEdit.status}
                                                onChange={this.onChangeEditable}
                                                defaultValue={status}
                                                name="priority"
                                                type="text"
                                            >
                                                {accessStatus.map(it => (
                                                    <Option key={it} value={it}>
                                                        {it}
                                                    </Option>
                                                ))}
                                            </Select>
                                        ) : null}
                                    </Descriptions.Item>
                                    <Descriptions.Item label="Приоритет">
                                        {modeControll === "default" ? (
                                            <Output className="priority">{priority}</Output>
                                        ) : modeControll === "edit" && modeControllEdit ? (
                                            <Select
                                                className="priorityEdit"
                                                value={modeControllEdit.priority}
                                                onChange={this.onChangeEditable}
                                                defaultValue={priority}
                                                name="priority"
                                                type="text"
                                            >
                                                {accessPriority.map(it => (
                                                    <Option key={it} value={it}>
                                                        {it}
                                                    </Option>
                                                ))}
                                            </Select>
                                        ) : null}
                                    </Descriptions.Item>
                                    <Descriptions.Item label="Автор задачи">
                                        <Output className="author">{author}</Output>
                                    </Descriptions.Item>
                                    <Descriptions.Item label="Исполнитель">
                                        {modeControll === "default" ? (
                                            <Output className="editor">
                                                {Array.isArray(editor) && editor.length === 1
                                                    ? editor
                                                    : Array.isArray(editor) && editor.length > 1
                                                    ? editor.join(",")
                                                    : null}
                                            </Output>
                                        ) : modeControll === "edit" && modeControllEdit ? (
                                            <Select
                                                className="editorEdit"
                                                value={modeControllEdit.editor}
                                                onChange={this.onChangeEditable}
                                                name="editor"
                                                mode="multiple"
                                                defaultValue={editor}
                                                placeholder="выберете исполнителя"
                                                optionLabelProp="label"
                                            >
                                                <Option value="Павел Петрович" label="Павел Петрович">
                                                    <span>Павел Петрович</span>
                                                </Option>
                                                <Option value="Гена Букин" label="Гена Букин">
                                                    <span>Гена Букин</span>
                                                </Option>
                                            </Select>
                                        ) : null}
                                    </Descriptions.Item>
                                    <Descriptions.Item label="Дата назначения">
                                        {modeControll === "default" ? (
                                            <Output className="startDate"> {date[0] ? date[0] : null}</Output>
                                        ) : modeControll === "edit" && modeControllEdit ? (
                                            <DatePicker
                                                value={moment(
                                                    modeControllEdit.date && modeControllEdit.date[0]
                                                        ? modeControllEdit.date[0]
                                                        : date[0]
                                                        ? date[0]
                                                        : moment(),
                                                    "DD.MM.YYYY"
                                                )}
                                                className="dateStartEdit"
                                                onChange={this.onChangeEditableStart}
                                                defaultValue={date[0] ? moment(date[0], "DD.MM.YYYY") : null}
                                                format="DD.MM.YYYY"
                                            />
                                        ) : null}
                                    </Descriptions.Item>
                                    <Descriptions.Item label="Дата завершения">
                                        {modeControll === "default" ? (
                                            <Output className="endDate"> {date[1] ? date[1] : null}</Output>
                                        ) : modeControll === "edit" && modeControllEdit ? (
                                            <DatePicker
                                                value={moment(
                                                    modeControllEdit.date && modeControllEdit.date[1]
                                                        ? modeControllEdit.date[1]
                                                        : date[1]
                                                        ? date[1]
                                                        : moment(),
                                                    "DD.MM.YYYY"
                                                )}
                                                className="dateEndEdit"
                                                onChange={this.onChangeEditableEnd}
                                                defaultValue={date[1] ? moment(date[1], "DD.MM.YYYY") : null}
                                                format="DD.MM.YYYY"
                                            />
                                        ) : null}
                                    </Descriptions.Item>
                                </Descriptions>
                                <div className="descriptionTask">
                                    <p className="descriptionTask__title">Задача</p>
                                    <p
                                        onClick={rulesEdit ? this.onEditContentMode : null}
                                        className={[
                                            "description",
                                            " descriptionTask__content",
                                            rulesEdit ? "editable" : null
                                        ].join(" ")}
                                    >
                                        {description ? description : "Описания задачи нету."}
                                        <span className="icon-wrapper">
                                            <i className="icon-pencil"></i>
                                        </span>
                                    </p>
                                    <p className="task_file">Дополнительные файлы для задачи</p>
                                    <File />
                                    <p className="descriptionTask__comment">Коментарии</p>
                                    <Comments rules={true} onUpdate={onUpdate} data={routeDataActive} />
                                </div>
                            </Scrollbars>
                        </div>
                        <div className="col-6 col-taskDescription">
                            <TitleModule classNameTitle="historyTaskTitle" title="Журнал работы" />
                            <Scrollbars>
                                {!jurnalDataKeys ? (
                                    <Empty description={<span>Нету данных в журнале</span>} />
                                ) : (
                                    this.renderWorkJurnal(jurnalDataKeys)
                                )}
                            </Scrollbars>
                        </div>
                    </div>
                </React.Fragment>
            );
        } else return <div>This task not found</div>;
    }
}

const mapStateTopProps = state => {
    return {
        router: state.router,
        publicReducer: state.publicReducer
    };
};

const mapDispatchToProps = dispatch => {
    return {
        onCaching: async (data, primaryKey, type, pk, store) =>
            await dispatch(middlewareCaching({ data, primaryKey, type, pk, store })),
        onUpdate: async (
            id,
            type,
            updateProp,
            updateFild,
            item,
            findStore,
            updateStore,
            multiply = false,
            limitUpdate
        ) =>
            await dispatch(
                middlewareUpdate({
                    id,
                    type,
                    updateProp,
                    updateFild,
                    item,
                    findStore,
                    updateStore,
                    multiply,
                    limitUpdate
                })
            )
    };
};

export { TaskView };
export default connect(mapStateTopProps, mapDispatchToProps)(TaskView);
