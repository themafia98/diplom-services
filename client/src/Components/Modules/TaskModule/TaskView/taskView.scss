.taskView {
    display: flex;

    .jurnalItem {
        display: flex;
        flex-direction: column;
        align-items: center;
        margin: 5px;
        border: 0.5px solid blue;
        background-color: #c7e1f7;
        border-radius: 6px;
        margin: 5px;
        .title {
            font-weight: bold;
        }
    }

    .ant-descriptions-row {
        .output {
            text-align: start;
        }
    }
    tbody {
        border: 0.5px solid $blueDark;
        tr {
            &.ant-descriptions-row {
                border-top: 0.5px solid $blueDark;
            }
            th {
                &.ant-descriptions-item-label {
                    border-right: 0.5px solid $blueDark;
                }
            }
            .ant-descriptions-item-content {
                max-width: 300px;
            }
        }
    }
    .historyTaskTitle {
        font-size: 23px;
        font-weight: bold;
        margin: 0;
        user-select: none;
        text-align: center;
    }
    .descriptionTask {
        .descriptionTask__title,
        .task_file,
        .descriptionTask__comment {
            text-align: center;
            font-weight: bold;
            padding: 5px;
            font-size: 16px;
        }
        .descriptionTask__content {
            display: block;
            width: 98%;
            min-height: 50px;
            max-height: 150px;
            font-size: 15px;
            padding: 10px;
            word-break: break-word;
            border: 1px solid $blueDark;
        }
    }
}

.col-taskDescription {
    height: 70vh;
    max-height: 70vh;
    .ant-descriptions {
        &.ant-descriptions-bordered {
            width: 98%;
        }
    }
}

.dropDownWrapper {
    max-width: 200px;
}

.ant-dropdown-trigger:hover {
    cursor: pointer;
    color: $blueDark;
}

.ant-dropdown-menu-item {
    cursor: pointer;
    p {
        text-align: center;
        cursor: pointer;
    }

    p:hover {
        cursor: pointer;
        color: red;
    }
}

.modalWindow {
    .ant-modal-content {
        .ant-modal-body {
            .ant-select {
                width: 100%;
            }
        }
    }
}
