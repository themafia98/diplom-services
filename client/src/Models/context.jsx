import { createContext } from 'react';
import config from '../config.json';
import Request from '../Models/Rest';
import Schema from '../Models/Schema';
import Notification from '../Models/NotificationRouter';
import { clientDB } from './ClientSideDatabase';

export const modelMethods = {
  clientDB,
  rest: new Request(),
  Request,
  config,
  schemaModel: Schema,
  schema: new Schema('no-strict'),
  Notification,
};

const modelsContext = createContext(modelMethods);

export default modelsContext;
