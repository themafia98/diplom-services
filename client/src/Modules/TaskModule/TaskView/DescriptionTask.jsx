import React, { forwardRef, useMemo } from 'react';
import clsx from 'clsx';
import Scrollbars from 'react-custom-scrollbars';

import Comments from 'Components/Comments';
import File from 'Components/File';

const DescriptionTask = forwardRef(
  (
    {
      onEditContentMode,
      onAddFileList,
      routeDataActive,
      onRemoveFile,
      commentProps,
      description,
      rulesEdit,
      filesArray,
      onUpdate,
      udata,
      rest,
      path,
    },
    ref,
  ) => {
    const scrollStyle = useMemo(() => {
      return { height: '150px' };
    }, []);
    return (
      <>
        <p className="descriptionTask__title">Задача</p>
        <div
          ref={ref}
          onDoubleClick={rulesEdit ? (event) => onEditContentMode(event, 'editDescription') : null}
          className={clsx('description', 'descriptionTask__content', rulesEdit ? 'editable' : null)}
        >
          <span className="icon-wrapper">
            <i className="icon-pencil" />
          </span>
          <Scrollbars style={scrollStyle}>
            <span className="descriptionContent">{description ? description : 'Описания задачи нету.'}</span>
          </Scrollbars>
        </div>

        <p className="task_file">Дополнительные файлы для задачи</p>
        <File
          filesArray={filesArray}
          rest={rest}
          onAddFileList={onAddFileList}
          onRemoveFile={onRemoveFile}
          moduleData={routeDataActive}
          isLocal={true}
          module="tasks"
        />
        <p className="descriptionTask__comment">Коментарии</p>
        <Comments
          commentProps={commentProps}
          udata={udata}
          path={path}
          rules={true}
          onUpdate={onUpdate}
          data={routeDataActive}
        />
      </>
    );
  },
);

export default DescriptionTask;
