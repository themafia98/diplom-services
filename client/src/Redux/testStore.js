import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

/** stores for  jest */

export const initialState = {
  tabReducer: {
    parentSize: null,
    childrenSize: null,
    flag: false,
  },
  publicReducer: {
    status: 'online',
    prewStatus: 'online',
    requestError: null,
    caches: {},
  },
  router: {
    currentActionTab: 'mainModule',
    actionTabs: ['MainModule'],
    routeDataActive: {},
    routeData: {},
  },
};

const middlewares = [thunk.withExtraArgument({})]; /** test middlewares */
const mockStore = configureStore(middlewares); /** test config store */

const store = mockStore(initialState); /** init test store */

export default store;
