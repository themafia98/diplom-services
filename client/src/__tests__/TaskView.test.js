import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { TaskView } from '../Modules/TaskModule/TaskView/TaskView';
import { initialState } from '../Redux/testStore';
import context from '../Models/context';
describe('<TaskView /> template', () => {
  test('Should be render with data', () => {
    const card = {
      _id: 'asdas',
      key: 'testTask',
      name: 'Task1',
      description: 'asdasd',
      status: 'В работе',
      priority: 'Средний',
      author: 'Павел Петрович',
      editor: ['Павел Петрович', 'Гена Букин'],
      comments: [],
      authorName: 'asdas',
      date: [],
    };
    const props = {
      onCaching: () => {},
      onUpdate: () => {},
      height: 800,
      data: { ...card },
      onLoadCurrentData: () => {},
      uuid: 'asd',
      rest: {},
      isBackground: false,
      visible: true,
      onLoadCacheData: () => {},
      udata: { _id: 'adasd', displayName: 'asdas' },
      modeControllEdit: { ...card },
      ...initialState,
      router: { ...initialState.router, routeDataActive: { ...card } },
    };
    const TaskViewWrapper = shallow(<TaskView {...props} />, { context });
    expect(toJson(TaskViewWrapper)).toMatchSnapshot();

    const keys = Object.keys(card);

    keys.forEach((keyCard) => {
      if (keyCard === 'description') {
      } else if (keyCard === 'date') {
        if (card.date[0]) expect(TaskViewWrapper.find('.startDate').exists()).toBeTruthy();

        if (card.date[1]) expect(TaskViewWrapper.find('.endDate').exists()).toBeTruthy();
      }
    });

    TaskViewWrapper.setState({ modeControll: 'edit' });
    expect(TaskViewWrapper.find('.statusEdit')).toBeTruthy();

    // if (card.description) expect(TaskViewWrapper.find('.description').simulate('click', [])).toEqual(0);
  });
});
