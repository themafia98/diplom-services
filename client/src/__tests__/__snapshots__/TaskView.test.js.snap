// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskView /> template Should be render with data 1`] = `
<Scrollbars
  autoHeight={false}
  autoHeightMax={200}
  autoHeightMin={0}
  autoHide={false}
  autoHideDuration={200}
  autoHideTimeout={1000}
  hideTracksWhenNotNeeded={false}
  renderThumbHorizontal={[Function]}
  renderThumbVertical={[Function]}
  renderTrackHorizontal={[Function]}
  renderTrackVertical={[Function]}
  renderView={[Function]}
  tagName="div"
  thumbMinSize={30}
  universal={false}
>
  <TitleModule
    additional=""
    className=""
    classNameTitle="taskModuleTittle"
    title="Карточка задачи"
  />
  <ModalWindow
    accessStatus={
      Array [
        "В работе",
        "Открыт",
        "Выполнен",
        "Закрыт",
      ]
    }
    actionType="__setJurnal"
    description="asdasd"
    editableContent="asdasd"
    key="testTask"
    keyTask="testTask"
    mode="jur"
    modeControll="default"
    modeEditContent={false}
    onCaching={[Function]}
    onCancelEditModeContent={[Function]}
    onEdit={[Function]}
    onRejectEdit={[Function]}
    onUpdate={[Function]}
    onUpdateEditable={[Function]}
    routeDataActive={
      Object {
        "author": "Павел Петрович",
        "date": Array [],
        "description": "asdasd",
        "editor": Array [
          "Павел Петрович",
          "Гена Букин",
        ],
        "key": "testTask",
        "name": "Task1",
        "priority": "Средний",
        "status": "В работе",
      }
    }
    statusTaskValue="В работе"
    typeRequst="POST"
    udata={Object {}}
  />
  <div
    className="taskView"
  >
    <div
      className="col-6 col-taskDescription"
    >
      <Scrollbars
        autoHeight={false}
        autoHeightMax={200}
        autoHeightMin={0}
        autoHide={false}
        autoHideDuration={200}
        autoHideTimeout={1000}
        hideTracksWhenNotNeeded={false}
        renderThumbHorizontal={[Function]}
        renderThumbVertical={[Function]}
        renderTrackHorizontal={[Function]}
        renderTrackVertical={[Function]}
        renderView={[Function]}
        tagName="div"
        thumbMinSize={30}
        universal={false}
      >
        <Descriptions
          bordered={true}
          column={
            Object {
              "d": 1,
              "lg": 1,
              "sm": 1,
              "xl": 1,
              "xs": 1,
              "xxl": 1,
            }
          }
          size="default"
        >
          <DescriptionsItem
            label="Артикул"
          >
            <Output
              action=""
              className="key"
              id=""
              typeOutput=""
            >
              testTask
            </Output>
          </DescriptionsItem>
          <DescriptionsItem
            label="Название"
          >
            <Output
              action=""
              className="name"
              id=""
              typeOutput=""
            >
              Task1
            </Output>
          </DescriptionsItem>
          <DescriptionsItem
            label="Статус"
          >
            <Output
              action=""
              className="status active"
              id=""
              typeOutput=""
            >
              В работе
            </Output>
          </DescriptionsItem>
          <DescriptionsItem
            label="Приоритет"
          >
            <Output
              action=""
              className="priority"
              id=""
              typeOutput=""
            >
              Средний
            </Output>
          </DescriptionsItem>
          <DescriptionsItem
            label="Автор задачи"
          >
            <Output
              action=""
              className="author"
              id=""
              typeOutput=""
            >
              Павел Петрович
            </Output>
          </DescriptionsItem>
          <DescriptionsItem
            label="Исполнитель"
          >
            <Output
              action=""
              className="editor"
              id=""
              typeOutput=""
            >
              Павел Петрович,Гена Букин
            </Output>
          </DescriptionsItem>
          <DescriptionsItem
            label="Дата назначения"
          >
            <Output
              action=""
              className="startDate"
              id=""
              typeOutput=""
            >
               
            </Output>
          </DescriptionsItem>
          <DescriptionsItem
            label="Дата завершения"
          >
            <Output
              action=""
              className="endDate"
              id=""
              typeOutput=""
            >
               
            </Output>
          </DescriptionsItem>
          <DescriptionsItem
            label="Затрачено времени"
          >
            <Output
              action=""
              id=""
              typeOutput=""
            >
              0.0 ч
            </Output>
          </DescriptionsItem>
        </Descriptions>
        <div
          className="descriptionTask"
        >
          <p
            className="descriptionTask__title"
          >
            Задача
          </p>
          <div
            className="description descriptionTask__content editable"
            onClick={[Function]}
          >
            <span
              className="icon-wrapper"
            >
              <i
                className="icon-pencil"
              />
            </span>
            <Scrollbars
              autoHeight={false}
              autoHeightMax={200}
              autoHeightMin={0}
              autoHide={false}
              autoHideDuration={200}
              autoHideTimeout={1000}
              hideTracksWhenNotNeeded={false}
              renderThumbHorizontal={[Function]}
              renderThumbVertical={[Function]}
              renderTrackHorizontal={[Function]}
              renderTrackVertical={[Function]}
              renderView={[Function]}
              style={
                Object {
                  "height": "150px",
                }
              }
              tagName="div"
              thumbMinSize={30}
              universal={false}
            >
              <span
                className="descriptionContent"
              >
                asdasd
              </span>
            </Scrollbars>
          </div>
          <p
            className="task_file"
          >
            Дополнительные файлы для задачи
          </p>
          <File
            filesArray={Array []}
            module="tasks"
            moduleData={
              Object {
                "author": "Павел Петрович",
                "date": Array [],
                "description": "asdasd",
                "editor": Array [
                  "Павел Петрович",
                  "Гена Букин",
                ],
                "key": "testTask",
                "name": "Task1",
                "priority": "Средний",
                "status": "В работе",
              }
            }
            onAddFileList={[Function]}
            onRemoveFile={[Function]}
            rest={Object {}}
          />
          <p
            className="descriptionTask__comment"
          >
            Коментарии
          </p>
          <Comments
            data={
              Object {
                "author": "Павел Петрович",
                "date": Array [],
                "description": "asdasd",
                "editor": Array [
                  "Павел Петрович",
                  "Гена Букин",
                ],
                "key": "testTask",
                "name": "Task1",
                "priority": "Средний",
                "status": "В работе",
              }
            }
            onUpdate={[Function]}
            rules={true}
            udata={Object {}}
          />
        </div>
      </Scrollbars>
    </div>
    <div
      className="col-6 col-taskDescription"
    >
      <TitleModule
        additional=""
        className=""
        classNameTitle="historyTaskTitle"
        title="Журнал работы"
      />
      <Scrollbars
        autoHeight={false}
        autoHeightMax={200}
        autoHeightMin={0}
        autoHide={false}
        autoHideDuration={200}
        autoHideTimeout={1000}
        hideTracksWhenNotNeeded={false}
        renderThumbHorizontal={[Function]}
        renderThumbVertical={[Function]}
        renderTrackHorizontal={[Function]}
        renderTrackVertical={[Function]}
        renderView={[Function]}
        tagName="div"
        thumbMinSize={30}
        universal={false}
      />
    </div>
  </div>
</Scrollbars>
`;
